csharp/source/autowrap/csharp/boilerplate.d

```diff
@@ -102,11 +104,45 @@ private string commonBoilerplate() @safe pure {
             }
         }
 
+        mixin(generateSharedTypes());
+
         public void pinPointer(void* ptr) nothrow {
             GC.setAttr(ptr, GC.BlkAttr.NO_MOVE);
             GC.addRoot(ptr);
         }
 
+        public datetime toDatetime(T)(T value)
+        if (isDateTimeType!T) {
+            return datetime(value);
+        }
+
+        public datetime[] toDatetime1DArray(T)(T[] value)
+        if (isDateTimeType!T) {
+            import std.algorithm : map;
+            import std.array : array;
+            return value.map!datetime.array;
+        }
+
+        public T fromDatetime(T)(datetime value)
+        if (isDateTimeType!T) {
+            static if (is(T == SysTime)) {
```

Comment:
Any particular reason you prefer an internal `static if` instead of one implementation per type?

Reply:
It is my understanding that like the other languages in the C family (C++/C#) D does not use return types when differentiating between overloads.

Reply:
What I meant was: why not

```d
T fromDatetime(T)(datetime value) if(is(Unqual!T == SysTime)) {
    // ...
}

T fromDatetime(T)(datetime value) if(is(Unqual!T == DateTime)) {
    // ...
}
```

instead?

Reply:
More compact? The compile down to effectively the same thing.

Reply:
I'd be surprised if it didn't compile to the same thing!

I don't know, Andrei has been vocal about this and I agree. After all, we use overloads instead of doing this:


```d
void fun(T)(T i) {
    static if(is(T == int)) {
        // ...
    } else static if(is(T == double) {
        // ...
    }
}
```

There's also the fact that using constraints affects which version gets picked. I just think it's cleaner overall.

Reply:
Done.